HW0/P4/P4.py

```diff
@@ -30,48 +35,142 @@
     # Load true trajectory and plot it
     # Normally, this data wouldn't be available in the real world
     #####################
+    s_true = pd.read_csv('./P4_trajectory.txt', header=None)
+    x_coords = s_true.loc[:, 0]
+    y_coords = s_true.loc[:, 1]
+    z_coords= s_true.loc[:, 2]
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '--b', label='True trajectory')
+    ax.plot(x_coords, y_coords, z_coords,
+             '--b', label='True trajectory')
 
     #####################
     # Part 2:
     #
     # Read the observation array and plot it (Part 2)
     #####################
+    measurements = pd.read_csv('./P4_measurements.txt', header=None).values # Keep it as a ndarray
+    # We now have to rescale x, y, and z appropriately. Luckily, the matrix
+    # is diagonal so we can just decouple everything.
+    rescaled_measurements = measurements.copy() #Copy...we don't want to overwrite the original measurements
+    rescaled_measurements[:, 0] *= 1./rx
+    rescaled_measurements[:, 1] *= 1./ry
+    rescaled_measurements[:, 2] *= 1./rz
+
+    rescaled_x = rescaled_measurements[:, 0]
+    rescaled_y = rescaled_measurements[:, 1]
+    rescaled_z = rescaled_measurements[:, 2]
 
-    # ax.plot(x_coords, y_coords, z_coords,
-    #         '.g', label='Observed trajectory')
+    ax.plot(rescaled_x, rescaled_y, rescaled_z,
+             '.g', label='Observed trajectory')
 
     #####################
     # Part 3:
     # Use the initial conditions and propagation matrix for prediction
     #####################
 
-    # A = ?
-    # a = ?
-    # s = ?
+    A = np.array([
+        [1, 0, 0, dt, 0, 0],
+        [0, 1, 0, 0, dt, 0],
+        [0, 0, 1, 0, 0, dt],
+        [0, 0, 0, 1-c*dt, 0, 0],
+        [0, 0, 0, 0, 1-c*dt, 0],
+        [0, 0, 0, 0, 0, 1-c*dt]
+    ], dtype=np.double)
```

Comment:
Instead of hard coding, you could also construct two arrays and use the np.diag() function to insert the arrays into a zero matrix. This should help if we are recording more parameters and have a huge A matrix. (This comment also applies to later matrix definitions, but I think it's a trivial point)